src/codeu/chat/server/Server.java

```diff
@@ -58,20 +61,57 @@ public Server(Uuid id, byte[] secret, Relay relay) {
 
     this.controller = new Controller(id, model);
     this.relay = relay;
-  }
 
-  public void syncWithRelay(int maxReadSize) throws Exception {
-    for (final Relay.Bundle bundle :  relay.read(id, secret, lastSeen, maxReadSize)) {
-      onBundle(bundle);
-      lastSeen = bundle.id();
-    }
+    timeline.scheduleNow(new Runnable() {
+      @Override
+      public void run() {
+        try {
+
+          LOG.info("Reading update from relay...");
+
+          for (final Relay.Bundle bundle : relay.read(id, secret, lastSeen, 32)) {
+            onBundle(bundle);
+            lastSeen = bundle.id();
+          }
+
+        } catch (Exception ex) {
+
+          LOG.error(ex, "Failed to read update from relay.");
+
+        }
+
+        // Do this all again in 60 seconds
+        timeline.scheduleIn(60000, this);
```

Comment:
I will add a TODO to add the hints - and will do that in a separate (<-- look I spelled it correctly this time) CL so that this change can stay focused on the Timeline.